app/views/concerns/entity_displaying_view.rb

```diff
@@ -205,14 +205,7 @@ def entity_death
   end
 
   def entity_date(dates)
-    return nil unless dates.present? && dates.is_a?(Array)
-
-    dates.each do |date|
-      gregorian = date_eras_gregorian(date)
-      return gregorian unless gregorian.nil?
-    end
-
-    dates.first.strip
+    date_most_accurate(dates)
```

Comment:
rather than calling it on the dates as returned, let's call it on the dates formated with `date_eras_gregorian` and call compact on that array.